src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2854,31 +2854,68 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
             if (targetArchitecture == TargetArchitecture.ARM && !_compilation.TypeSystemContext.Target.IsWindows)
                 flags.Set(CorJitFlag.CORJIT_FLAG_RELATIVE_CODE_RELOCS);
 
-            if ((targetArchitecture == TargetArchitecture.X86
-                || targetArchitecture == TargetArchitecture.X64)
+            if (targetArchitecture == TargetArchitecture.X86)
+            {
+                flags.Set(InstructionSet.X86_SSE);
+                flags.Set(InstructionSet.X86_SSE2);
+#if !READYTORUN
+                // This list needs to match the list of intrinsics we can generate detection code for
+                // in HardwareIntrinsicHelpers.EmitIsSupportedIL.
+#else
+                // For ReadyToRun, this list needs to match up with the behavior of FilterNamedIntrinsicMethodAttribs
+                // In particular, that this list of supported hardware will not generate non-SSE2 safe instruction
+                // sequences when paired with the behavior in FilterNamedIntrinsicMethodAttribs
+                if (isMethodDefinedInCoreLib())
+#endif
+                {
+                    flags.Set(InstructionSet.X86_AES);
+                    flags.Set(InstructionSet.X86_PCLMULQDQ);
+                    flags.Set(InstructionSet.X86_SSE3);
+                    flags.Set(InstructionSet.X86_SSSE3);
+                    flags.Set(InstructionSet.X86_LZCNT);
```

Comment:
I know this is mirroring the current state, but LZCNT was released the same time as AVX2, BMI1, and BMI2. I don't think it makes sense to have it here.

Reply:
I *REALLY* don't want to change any of the behavior of the system with this change. Also, fortunately, there isn't actually any use of this instruction in corelib, so it doesn't really matter.

If we want to change the current state of the world, we should do it in another PR.

Reply:
> If we want to change the current state of the world, we should do it in another PR.

I'm fine with that, but we definitely are using it from multiple places in corelib: https://source.dot.net/#System.Private.CoreLib/BitOperations.cs,f10c861b0040aaea,references